framework/src/functions/MooseParsedFunctionWrapper.C

```diff
@@ -92,32 +98,40 @@ MooseParsedFunctionWrapper::initialize()
   for (unsigned int i = 0; i < _vals_input.size(); ++i)
   {
     // Case when a Postprocessor is found by the name given in the input values
-    if (_feproblem.hasPostprocessor(_vals_input[i]))
+    if (_feproblem_ptr->hasPostprocessor(_vals_input[i]))
     {
-      Real & pp_val = _feproblem.getPostprocessorValue(_vals_input[i]);
+      Real & pp_val = _feproblem_ptr->getPostprocessorValue(_vals_input[i]);
       _initial_vals.push_back(pp_val);
       _pp_vals.push_back(&pp_val);
       _pp_index.push_back(i);
     }
 
     // Case when a scalar variable is found by the name given in the input values
-    else if (_feproblem.hasScalarVariable(_vals_input[i]))
+    else if (_feproblem_ptr->hasScalarVariable(_vals_input[i]))
     {
-      Real & scalar_val = _feproblem.getScalarVariable(_tid, _vals_input[i]).sln()[0];
+      Real & scalar_val = _feproblem_ptr->getScalarVariable(_tid, _vals_input[i]).sln()[0];
       _initial_vals.push_back(scalar_val);
       _scalar_vals.push_back(&scalar_val);
       _scalar_index.push_back(i);
     }
 
     // Case when a function is found by the name given in the input values
-    else if (_feproblem.hasFunction(_vals_input[i]))
+    else if (_feproblem_ptr->hasFunction(_vals_input[i]))
     {
-      Function & fn = _feproblem.getFunction(_vals_input[i], _tid);
+      Function & fn = _feproblem_ptr->getFunction(_vals_input[i], _tid);
       _initial_vals.push_back(0);
       _functions.push_back(&fn);
       _function_index.push_back(i);
     }
 
+    // Error when a variable is supplied
+    else if (_feproblem_ptr->hasVariable(_vals_input[i]))
+      _moose_object_ptr->paramError(
+          "vals",
+          "The 'vals' input includes an item ('",
+          _vals_input[i],
+          "') that is a variable, which is not supported by the ParsedFunction object.");
+
```

Comment:
I guess this is the meat of this PR (not sure why all the other changes are there). Why didn't this print the error message from the last `else` branch? It explicitly references scalar variables. Why did it result in an MPI_ABORT with no useful error? It seems to me that there might be an underlying issue that needs to get fixed.

Reply:
I'm with @dschwen here. I'd prefer to keep references in this object instead of switching everything to pointers. You are using the checkedPointer function, just dereference there and leave the rest of the code the same. Then there's no question throughout as to whether the pointer ever needs to be checked for validity.